internal/services/azapi_resource.go

```diff
@@ -929,6 +938,40 @@ func expandBody(body map[string]interface{}, model AzapiResourceModel) diag.Diag
 	return diag.Diagnostics{}
 }
 
+func preflightValidation(ctx context.Context, client *clients.ResourceClient, requestPlan *AzapiResourceModel, body map[string]interface{}) error {
```

Comment:
Please use `id parse.ResourceId` instead of `requestPlan *AzapiResourceModel`, because the necessary information could be retrieved by the parse.ResourceId.

Reply:
updated. Thanks.